docs/Deploy-Module.md

```diff
@@ -1,39 +1,96 @@
+
+[Ansible](http://docs.ansible.com), here is the [download link](http://docs.ansible.com/intro_installation.html)
+
+Ansible is a pure python package, so you need to have python installed, follow the docs in the link above. 
+
+### What is Ansible?
+
+Ansible is a language and toolset to define how to provision(install software, configure the system). It allows you to manipulate remote servers on your laptop, any number of nodes in parallel!
+
+### Why Use Ansible?
+
+When set up a Tachyon cluster, we need the ability to manipulate target machines in parallel, say, install java on all nodes. Ansible satisfies this requirement. It's supported by Vagrant and has simple syntax, so we adopt it as the provisioning tool.
+
+If you want to deploy on virtualbox on your laptop, please install [Virtualbox](https://www.virtualbox.org/wiki/Downloads).
+
+## Quick Start With VirtualBox
+
+After installing the prerequisites, if have virtualbox installed, you can follow this quick start to set up a two node Tachyon cluster with apache hadoop2.4.1 as under filesystem, one node is named TachyonMaster, another is TachyonWorker.
+
+1. `cd TACHYON/deploy/vagrant`, replace TACHYON with the Tachyon repo on your host, you can clone the latest from [github](https://github.com/amplab/tachyon.git)
+2. `./run_vb.sh`, it will be slow when you first run this command, because vagrant will download a centos6.5 virtual machine, and install necessary software in this machine. The console output is verbose enough for you to know what's going on. 
+3. After the above command, a purple line in the output tells you the IP of TachyonMaster, the IP should be "192.168.1.12", visit `http://192.168.1.12:19999` in your browser, you should see Tachyon's web UI
+4. Visit `http://192.168.1.12:50070` in your browser, you should see Hadoop's web UI!
+5. `vagrant ssh TachyonMaster`, and you should have entered the virtual machine named TachyonMaster
+6. `/tachyon/bin/tachyon runTests` to run some tests against Tachyon
+7. After the tests all pass, visit Tachyon web UI: `http://192.168.1.12:19999` again, in `Browse FileSystem`, you should see files written to Tachyon by the tests run above
+8. If you don't want to play around in the cluster any more, `vagrant destory` to delete the virtual machines. 
+
+Cool! The more exciting aspect of deploy module is that you can set up the same cluster in AWS EC2 with the same commands and workflow described above, just change some configurations!
+
+## Configurations
+
+There are three kinds of configuration files under `conf`. All the configuration files use [yaml syntax](http://en.wikipedia.org/wiki/YAML).
+
+1. configuration of virtual machine itself in `init.yml`. 
+2. provider specific configuration, like `ec2.yml`, `openstack.yml`.
+3. software configuration, like `tachyon.yml`, `spark.yml`, `ufs.yml`(ufs means under filesystem). 
+
+Meanings of fields in the configurations are explained in detail in the yml files.
+
+## AWS EC2
```

Comment:
Is there any info on the ec2-cli commandline tools? I think that should be somewhere...